src/CacheMiddleware.php

```diff
@@ -113,6 +113,10 @@ public function __invoke(callable $handler)
         return function (RequestInterface $request, array $options) use (&$handler) {
             if (!isset($this->httpMethods[strtoupper($request->getMethod())])) {
                 // No caching for this method allowed
+
+                // Invalidate cache after a non-safe method on the same URI will be invalidated
+                $this->cacheStorage->delete($request);
```

Comment:
> A cache MUST invalidate the effective Request URI (Section 5.5 of
> [RFC7230]) as well as the URI(s) in the Location and Content-Location
> response header fields (if present) when a non-error status code is
> received in response to an unsafe request method.
> [...]
> Here, a "non-error response" is one with a 2xx (Successful) or 3xx 
> (Redirection) status code.
https://tools.ietf.org/html/rfc7234#section-4.4

I thing that it's better if this action is done after checking that the response are "non-error". Can we move it in the `then(...)` just below?

Reply:
> > A cache MUST invalidate the effective Request URI (Section 5.5 of
> > [RFC7230]) as well as the URI(s) in the Location and Content-Location
> > response header fields (if present) when a non-error status code is
> > received in response to an unsafe request method.
> > [...]
> > Here, a "non-error response" is one with a 2xx (Successful) or 3xx
> > (Redirection) status code.
> > https://tools.ietf.org/html/rfc7234#section-4.4
> 
> I thing that it's better if this action is done after checking that the response are "non-error". Can we move it in the `then(...)` just below?

Makes sense.